File: src/main/java/org/cerion/stocklist/charts/CandleDataSet.java
package org.cerion.stocklist.charts;

import org.cerion.stocklist.PriceList;

public class CandleDataSet implements IDataSet {

    private PriceList mList;
    private String mLabel;
    private int mLabelColor;

    protected CandleDataSet(PriceList list, String label, int labelColor) {
        mList = list;
        mLabel = label;
        mLabelColor = labelColor;
    }

    @Override
    public int size() {
        return mList.size() - 1;
    }

    @Override
    public LineType getLineType() {
        return LineType.CANDLE;
    }

    @Override
    public String getLabel() {
        return mLabel;
    }

    public int getColor() {
        return mLabelColor;
    }

    public float getHigh(int pos) {
        return mList.mHigh.get(pos+1);
    }

    public float getLow(int pos) {
        return mList.mLow.get(pos+1);
    }

    public float getOpen(int pos) {
        return mList.mOpen.get(pos+1);
    }

    public float getClose(int pos) {
        return mList.close(pos+1);
    }


}

File: src/main/kotlin/charts/CandleDataSet.kt
package org.cerion.stocklist.charts

import org.cerion.stocklist.PriceList

class CandleDataSet(private val mList: PriceList, override val label: String, override val color: Int) : IDataSet {

    override val size: Int = mList.size - 1
    override val lineType: LineType = LineType.CANDLE

    fun getHigh(pos: Int): Float = mList.mHigh[pos + 1]
    fun getLow(pos: Int): Float = mList.mLow[pos + 1]
    fun getOpen(pos: Int): Float = mList.mOpen[pos + 1]
    fun getClose(pos: Int): Float = mList.close(pos + 1)
}
